website/addons/wiki/static/wikiPage.js

```diff
@@ -0,0 +1,262 @@
+                    }
+                    self.displaySource(rawContent);
+                });
+            }
+        } else {
+            self.displaySource('');
+        }
+    });
+}
+
+    // currentText comes from ViewWidget.displayText
+function CompareWidget(visible, compareVersion, currentText, rendered, contentURL) {
+    var self = this;
+
+    self.compareVersion = compareVersion;
+    self.currentText = currentText;
+    self.rendered = rendered;
+    self.visible = visible;
+    self.contentURL = contentURL;
+    self.compareSource = ko.observable('');
+
+    self.compareText = ko.computed(function() {
+        var requestURL;
+        if (self.compareVersion() === 'current') {
+            requestURL = self.contentURL;
+        } else {
+            requestURL= self.contentURL + self.compareVersion();
+        }
+        var request = $.ajax({
+            url: requestURL
+        });
+        request.done(function (resp) {
+            var rawText = resp.wiki_content;
+            self.compareSource(rawText);
+        });
+
+    });
+
+    self.compareOutput = ko.computed(function() {
+        var output = diffTool.diff(self.compareSource(), self.currentText());
+        self.rendered(output);
+        return output;
+    }).extend({ notify: 'always' });
+
+}
+
+
+var defaultOptions = {
+    editVisible: false,
+    viewVisible: true,
+    compareVisible: false,
+    menuVisible: true,
+    canEdit: true,
+    viewVersion: 'current',
+    compareVersion: 'previous',
+    urls: {
+        content: '',
+        draft: '',
+        page: ''
+    },
+    metadata: {}
+};
+
+function ViewModel(options){
+    var self = this;
+
+    // enabled?
+    self.editVis = ko.observable(options.editVisible);
+    self.viewVis = ko.observable(options.viewVisible);
+    self.compareVis = ko.observable(options.compareVisible);
+    self.menuVis = ko.observable(options.menuVisible);
+
+    self.compareVersion = ko.observable(options.compareVersion);
+    self.viewVersion = ko.observable(options.viewVersion);
+    self.draftURL = options.urls.draft;
+    self.contentURL = options.urls.content;
+    self.pageURL = options.urls.page;
+    self.editorMetadata = options.metadata;
+    self.canEdit = options.canEdit;
+
+    self.viewText = ko.observable('');
+    self.renderedView = ko.observable('');
+    self.renderedCompare = ko.observable('');
+    self.allowMathjaxification = ko.observable(true);
+    self.allowFullRender = ko.observable(true);
+
+    self.currentURL = ko.computed(function() {
+        var url = self.pageURL;
+
+        // Default view is special cased
+        if (!self.editVis() && self.viewVis() && self.viewVersion() === 'current' && !self.compareVis() && self.menuVis()) {
+            history.replaceState({}, '', url);  // jshint ignore: line
+            return;
+        }
+
+        var paramPrefix = '?';
+
+        if (self.editVis()) {
+            url += paramPrefix + 'edit';
+            paramPrefix = '&';
```

Comment:
Talked to @sloria; we are going to defer this because using JQuery param would make messier URLs.